api/app/Generators/UserExcelGenerator.php

```diff
@@ -142,140 +224,668 @@ public function generate(): self
+        }
+
+        return '';
+    }
+
+    /**
+     * Get featured skill justification from model
+     */
+    private function getJustificationFromExperienceSkill($experience, $userSkill): string
+    {
+        if (empty($experience->id) || empty($userSkill->id)) {
+            return '';
+        }
+
+        $experienceSkill = \App\Models\ExperienceSkill::where('experience_id', $experience->id)
+            ->where('experience_type', get_class($experience))
+            ->where('user_skill_id', $userSkill->id)
+            ->first();
+
+        return $experienceSkill->details ?? '';
+    }
+
+    /**
+     * Get all possible skill names for a skill key
+     */
+    private function getSkillNames(string $featuredSkillKey): array
+    {
+        $klcSkillNames = [
+            'achieve_results' => [
+                'en' => 'Achieve Results',
+                'fr' => 'Obtenir des résultats',
+            ],
+            'character_leadership' => [
+                'en' => 'Character Leadership',
+                'fr' => 'Leadership de caractère',
+            ],
+            'collaborate_with_partners_and_stakeholders' => [
+                'en' => 'Collaborate with Partners and Stakeholders',
+                'fr' => 'Collaborer avec les partenaires et les intervenants',
+            ],
+            'create_vision_and_strategy' => [
+                'en' => 'Create Vision and Strategy',
+                'fr' => 'Créer une vision et une stratégie',
+            ],
+            'mobilize_people' => [
+                'en' => 'Mobilize People',
+                'fr' => 'Mobiliser les personnes',
+            ],
+            'promote_innovation_and_guide_change' => [
+                'en' => 'Promote Innovation and Guide Change',
+                'fr' => 'Promouvoir l\'innovation et orienter le changement',
+            ],
+            'uphold_integrity_and_respect' => [
+                'en' => 'Uphold Integrity and Respect',
+                'fr' => 'Préserver l\'intégrité et le respect',
+            ],
+        ];
+
+        $skillNames = $klcSkillNames[$featuredSkillKey] ?? [];
+        $currentLangName = trim($skillNames[$this->lang] ?? '');
+
+        return ! empty($currentLangName) ? [$currentLangName] : [];
+    }
+
+    /**
+     * Calculate number of months between dates
+     */
+    private function calculateMonths($startDate, $endDate): int
+    {
+        if (! $startDate) {
+            return 0;
+        }
+
+        if (! $endDate) {
+            $endDate = now();
+        }
+
+        return $startDate->diffInMonths($endDate);
+    }
+
+    /**
+     * Get users ids from query
+     */
+    private function getUserIdsFromQuery(): array
+    {
+        $userIds = [];
+        $query = $this->buildQuery();
+
+        $query->chunk(200, function ($users) use (&$userIds) {
+            foreach ($users as $user) {
+                $userIds[] = $user->id;
+            }
+        });
+
+        return $userIds;
+    }
```

Comment:
Looks as though we call two DB queries to get users in this. If we always build the user rows first, I wonder if this is better tracked as a private property?

```php
private array $userIds = [];

private function generateUsersSheet()
{
  // ...
   $query->chunk(200, function ($users) use ($sheet, &$currentUser) {
     foreach ($users as $user) {
       $this->userIds[] = $user->id;
     }
   });
  //...
}
```